lib/rubocop/cop/rspec/file_path.rb

```diff
@@ -82,30 +82,39 @@ def on_top_level_example_group(node)
         private
 
         def ensure_correct_file_path(send_node, described_class, arguments)
-          glob = glob_for(described_class, arguments.first)
-          return if filename_ends_with?(glob)
-
-          add_offense(send_node, message: format(MSG, suffix: glob))
+          pattern = pattern_for(described_class, arguments.first)
+          return if filename_ends_with?(pattern)
+
+          # For the suffix shown in the offense message, modify the regular
+          # expression pattern to resemble a glob pattern for clearer error
+          # messages.
+          offense_suffix = pattern.gsub('.*', '*').sub('[^/]', '')
+            .sub('\.', '.')
+          add_offense(send_node, message: format(MSG, suffix: offense_suffix))
         end
 
         def routing_spec?(args)
           args.any?(&method(:routing_metadata?))
         end
 
-        def glob_for(described_class, method_name)
-          return glob_for_spec_suffix_only? if spec_suffix_only?
+        def pattern_for(described_class, method_name)
+          return pattern_for_spec_suffix_only? if spec_suffix_only?
 
-          "#{expected_path(described_class)}#{name_glob(method_name)}*_spec.rb"
+          [
+            expected_path(described_class),
+            name_pattern(method_name),
+            '[^/]*_spec\.rb'
+          ].join
         end
 
-        def glob_for_spec_suffix_only?
-          '*_spec.rb'
+        def pattern_for_spec_suffix_only?
+          '.*_spec.rb'
```

Comment:
Sorry I hadn’t noticed this one before.

```suggestion
          '.*_spec\.rb'
```

Reply:
This is now resolved and I added specs for it.